wab/src/main/java/org/apache/unomi/web/ContextServlet.java

```diff
@@ -136,11 +136,17 @@ public void service(ServletRequest request, ServletResponse response) throws Ser
             try {
                 contextRequest = mapper.readValue(factory.createParser(stringPayload), ContextRequest.class);
             } catch (Exception e) {
+                ((HttpServletResponse)response).sendError(HttpServletResponse.SC_BAD_REQUEST, "Check logs for more details");
```

Comment:
Shouldn't the HTTP code be more like 500 Internal Server Error? After all we don't know if the request is bad or we have another issue in our code / infra. Or perhaps it would be better to distinguish the user input errors (JsonParseException and JsonMappingException for example, didn't check in details) and respond with 400, otherwise respond with 500.

Reply:
The ticket is only aboute the post request, i asked to Serge.

Reply:
I think you meant to respond to my below comment and not this one, is that right? Could you also clarify what was his exact comment / explanation what are you trying to fix and put it in the JIRA ticket. Do we want to forbit all POST requests on this resource?

Another thing - as far as I know we create an UNOMI ticket with all the necessary details and reference it in commit message - could you do it please?

Reply:
I don't understand what you are talking about @loomchild . The modifications in this commit only intend to make sure that an HTTP error code is sent when an error is detect. Before these changes the error code was not returned and the default code (200) was being sent back, which is exactly what we are trying to solve. But I do agree a ticket in the UNOMI JIRA should have been created first and used as a reference in this PR title. Please correct this @dinesh1996

Reply:
Here I am suggesting to perhaps respond with HTTP 400 Bad Request just when the error is caused by user request input data (catch just JsonParseException and JsonMappingException for example) and respond with HTTP 500 Internal Server Error in case of any other exception (NPE, IOException - since mapper.readValue() can throw it I suppose it communicates with something external).

I am not sure if this is a valid suggestion, since I don't know the code well yet. Perhaps we *want* to treat all errors in this line as HTTP 400 Bad request since they are always caused by used data (waiting for your confirmation):
`contextRequest = mapper.readValue(factory.createParser(stringPayload), ContextRequest.class);`

Reply:
Ok that makes sense to me but it's not related to the HTTP method at all, rather the actual code executing in the context of an HTTP call. However this is extending the initial scope of this modification and I suggest we create a seperate issue for this so that we can access this already useful change.